File: Accounting_frontend/src/dictionaries/model/dictionary-element-base.ts
import { ModelElementBase } from 'src/model/model-element-base';

/**
 * Базовый класс элементов-справочников модели данных
 */
export class DictionaryElementBase extends ModelElementBase {
  /**
   * Признак удаления
   */
  public removed: boolean = false;

  /**
   * Дата удаления
   */
  public removedDate: Date = new Date();
}

File: AccountingDAL/Model/Dictionaries/DictionaryElementBase.cs
namespace AccountingDAL.Model.Dictionaries
{
    /// <summary>
    /// Базовый класс элементов-справочников модели данных
    /// при удалении элементы не удаляются из базы а проставляются поля Removed и RemovedDate
    /// </summary>
    public abstract class DictionaryElementBase : ModelElementBase
    {
        /// <summary>
        /// Признак удаления
        /// </summary>
        public bool Removed { get; set; } = false;

        /// <summary>
        /// Дата удаления
        /// </summary>
        public DateTime RemovedDate { get; set; } = DateTime.MinValue;
    }
}
